src/rgw/rgw_trim_bilog.cc

```diff
@@ -571,13 +735,40 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
       return set_cr_error(retcode);
     }
 
+    retcode = maybe_remove_generation();
+    if (retcode < 0) {
+      ldpp_dout(dpp, 4) << "remove removing old generation from log: "
+			<< cpp_strerror(retcode) << dendl;
+      return set_cr_error(retcode);
+    }
+
+    if (clean_info) {
+      yield call(new RGWPutBucketInstanceInfoCR(
+		   store->svc()->rados->get_async_processor(),
+		   store, clean_info->first, false, {},
+		   nullptr, dpp));
+      if (retcode < 0) {
+	ldpp_dout(dpp, 0) << "failed to put bucket info: "
+			  << cpp_strerror(retcode) << dendl;
+	return set_cr_error(retcode);
+      }
+      yield call(new BucketCleanIndexCollectCR(dpp, store, clean_info->first,
+					       clean_info->second));
```

Comment:
All right. I wasn't sure if having the log thee after we deleted some shards would cause problems.

Reply:
Fixed.

Reply:
> All right. I wasn't sure if having the log thee after we deleted some shards would cause problems.

yeah, we might get some unexpected ENOENT errors, but hopefully not in the normal sync path because no peers should be requesting this old generation anymore